internal/dag/builder_test.go

```diff
@@ -1107,6 +1139,193 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 				},
 			),
 		},
+		"different weights for multiple forwardTos": {
+			gateway: gatewayWithSelector,
+			objs: []interface{}{
+				kuardService,
+				kuardService2,
+				kuardService3,
+				&gatewayapi_v1alpha1.HTTPRoute{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "basic",
+						Namespace: "projectcontour",
+						Labels: map[string]string{
+							"app":  "contour",
+							"type": "controller",
+						},
+					},
+					Spec: gatewayapi_v1alpha1.HTTPRouteSpec{
+						Rules: []gatewayapi_v1alpha1.HTTPRouteRule{{
+							Matches: httpRouteMatch(gatewayapi_v1alpha1.PathMatchPrefix, "/"),
+							ForwardTo: []gatewayapi_v1alpha1.HTTPRouteForwardTo{
+								{
+									ServiceName: pointer.StringPtr("kuard"),
+									Port:        gatewayPort(8080),
+									Weight:      5,
+								}, {
+									ServiceName: pointer.StringPtr("kuard2"),
+									Port:        gatewayPort(8080),
+									Weight:      10,
+								},
+								{
+									ServiceName: pointer.StringPtr("kuard3"),
+									Port:        gatewayPort(8080),
+									Weight:      15,
+								},
+							},
+						}},
+					},
+				},
+			},
+			want: listeners(
+				&Listener{
+					Port: 80,
+					VirtualHosts: virtualhosts(
+						virtualhost("*", prefixroute("/",
+							&Service{
+								Weighted: WeightedService{
+									Weight:           5,
+									ServiceName:      kuardService.Name,
+									ServiceNamespace: kuardService.Namespace,
+									ServicePort:      kuardService.Spec.Ports[0],
+								},
+							},
+							&Service{
+								Weighted: WeightedService{
+									Weight:           10,
+									ServiceName:      kuardService2.Name,
+									ServiceNamespace: kuardService2.Namespace,
+									ServicePort:      kuardService2.Spec.Ports[0],
+								},
+							},
+							&Service{
+								Weighted: WeightedService{
+									Weight:           15,
+									ServiceName:      kuardService3.Name,
+									ServiceNamespace: kuardService3.Namespace,
+									ServicePort:      kuardService3.Spec.Ports[0],
+								},
+							},
+						)),
+					),
+				},
+			),
+		},
+		"one service weight zero w/weights for other forwardTos": {
```

Comment:
It's an easy way to stage changes to routes. So you can configure the route with a new service of weight 0, then add the service and update the weight. 

For Contour, it's nice because the clusters/endpoints will be configured in Envoy, then to activate the traffic can flow faster without needing configuration changes to push over to Envoy.

Reply:
👍🏽 that makes sense, switching weights to switch over to a new backend